js/lib/appUrlUtil.js

```diff
@@ -24,6 +24,13 @@ module.exports.fileUrl = (filePath) => {
   return encodeURI('file://' + fileUrlPath)
 }
 
+module.exports.chromeUrl = (filePath = '') => {
+  filePath = module.exports.fileUrl(filePath)
+  filePath = filePath.replace('file://', 'chrome://brave')
```

Comment:
where is the `chrome://brave` to `file://` mapping implemented in muon? it seems like a potential security hole since it can be used to bypass CSP restrictions; we should make sure `chrome://brave` loads are blocked in untrusted contexts.

Reply:
also i'm curious what CSP violations are being triggered? `app/extensions/brave/index.html` explicitly allows `img-src file:` but not `chrome:` so i'm confused why the original issue happens

Reply:
I played with the CSP file but couldn't get the violation to reveal itself either. It might be some other security restriction. The error from the console is

    [75566:775:0922/145254.257520:ERROR:CONSOLE(0)] "Not allowed to load local resource:
        file:///Users/kevin/Desktop/brave/browser-laptop-bootstrap/src/browser-laptop/app/extensions/metalmash/images/icon-19.png", 
        source: chrome://brave/Users/kevin/Desktop/brave/browser-laptop-bootstrap/src/browser-laptop/app/extensions/brave/index-dev.html (0)

and it disappears when switching to `chrome://brave`. Subsequently the image loads.

Reply:
@diracdeltas chrome://brave can't be loaded from web pages `Not allowed to load local resource: chrome://brave/usr/local/brave/browser-laptop-bootstrap/README.md`. When I originally tested them it also wasn't possible to load them by entering the url manually, but I just tested again and it is now possible because we added support for `chrome://*` urls in general so we should probably block them from the urlbar

Reply:
Also, the csp can't allow something that is prohibited by default (access to file resources) which is why the file:// urls don't work

Reply:
I think it's fine if chrome://brave  URLs can be loaded manually from the urlbar since that is also possible for file URLs. just want to make sure there is nothing they can do that files URLs can't do, other than be loaded in the chrome:// context.